### DISCUSSION TITLE
Adjust firewall rule in MSI package?

### DISCUSSION THREAD

**Discussion by queuedvariable** [PowerShell/Win32-OpenSSH#2203] [Q&A]
Hi, 

I'm a unix native trying to manage some windows machines and have built out AD and WSUS and such and begun to roll this SSH package out.   It's really great and powershell really makes the whole ordeal much more bearable but I notice the firewall exception the MSI package installs only opens port 22 for sshd.exe to the 'private' zone.    Can anybody point me to a resource where I can figure out how to edit the MSI to open it to all the zones (or at least domain too)?      Maybe with Orca or something?

thanks,

-qv

**Comment by maertendMSFT** [comment]
The simple solution that we recommend is to wrap the MSI in in a PowerShell script that will modify the firewall rules after creation.  
Check out this doc on changing the firewall rule, specifically the `profile` parameter: https://learn.microsoft.com/powershell/module/netsecurity/set-netfirewallrule?view=windowsserver2022-ps#-profile

These are the commands that we recommend depending on the desired scope:
- `Set-NetFirewallRule -Name sshd -Profile Domain`
- `Set-NetFirewallRule -Name sshd -Profile Public`

Otherwise, you will need to create your own fork of the repo and build your own version of the MSI.
